net: Set Dummy net_if as default one for tests

jukkar opened the pull request:
If we run network tests under tests/net, then set the Dummy network interface as a default one so that it will be picked up first.

This should solve the issue if the DUT is having a real network interfaces like onboard Ethernet interface which could cause the test to fail. The test might fail in this case because the network tests assume that only simulated network interfaces are used by the tests.

jukkar committed:
net: Set Dummy net_if as default one for tests

If we run network tests under tests/net, then set the
Dummy network interface as a default one so that it will
be picked up first.

This should solve the issue if the DUT is having a real
network interfaces like onboard Ethernet interface which
could cause the test to fail. The test might fail in this
case because the network tests assume that only simulated
network interfaces are used by the tests.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar commented:
@hakehuang please try this one if it solves the issues you see in https://github.com/zephyrproject-rtos/zephyr/issues/64062

rlubos commented on subsys/net/ip/Kconfig:
```diff
@@ -839,9 +839,13 @@ config NET_HEADERS_ALWAYS_CONTIGUOUS
 choice
 	prompt "Default Network Interface"
 	default NET_DEFAULT_IF_FIRST
+	default NET_DEFAULT_IF_DUMMY if NET_TEST
```
This should go before `default NET_DEFAULT_IF_FIRST`, otherwise it won't have effect. But I can confirm, that on `nucleo_h723zg` the tests passed with this change :+1:

jukkar committed:
tests: net: iface: Do not use eth0 name in the test

If the first Ethernet interface is not the first network
interface, then there might be two interfaces with the same name.
The test does not like that.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>
